### ISSUE TITLE
vmagent API endpoint to programmatically test relabeling rules

### ISSUE THREAD

**Issue by hartfordfive** [VictoriaMetrics/VictoriaMetrics#6821]
### Is your feature request related to a problem? Please describe

From what I can tell, it's not currently possible to submit a list of relabeling rules and target labels to then receive a simple JSON response with the final computed labels.  

### Describe the solution you'd like

The vmagent `/metric-relabel-debug` UI endpoint can be used to debug relabeling rules although it's not practical when performing systematic tests.   For example, I would like to add unit tests for relabeling rules for various scrape jobs and have the ability in the CICD pipeline to validate the resulting target labels, given a set of relabeling rules and input labels, matches the expectation.   I currently have a use-case where I need to add multiple relabeling rules in an already large list of relabeling rules, thus it would be great to have the ability to pass in pre-defined sets of metrics and test them against the current relabeling rules.

For example, the could be a `/api/v1/metric-relabel-debug` endpoint, accepting POST, exposed by vmagent which could receive formdata same as the UI:
```
relabel_configs: <RELABEL_CONFIG_RULES>
metric: {<METRIC_LABEL_VALUE_PAIRS>}
```
but return a simple JSON body:
```json
{
  "status": "OK"
  "labels": {
    "label1": "value1",
    "label2": "value2",
    "label3": "value3",
  }
```
The UI could also simply be updated to accept a `response_format` query string argument, and return only a basic JSON response given the value of the argument is `json`.

### Describe alternatives you've considered

Could potentially use the web interface version although to would require some html scraping to properly extract the desired output.

**Comment by zekker6** [comment]
Helllo @hartfordfive 
VMUI already uses such API endpoint - `select/{tenant}/prometheus/metric-relabel-debug` .
This API endpoint requires the following query args:
- `format` = `json` - in order to get response in JSON format
- `relabel_configs` 
- `metric`
So you can integrate it into your workflow.

You can check the format for this endpoint here - https://play.victoriametrics.com/select/0/vmui/?#/relabeling
There is a "try example" button which has some pre-defined content for both relabel configs and metrics. It's possible to open dev tools network tab to see the content passing through.

**Comment by hartfordfive** [comment]
I would like to be able to test both the relabel_configs and metric_relabel_configs sections in a single test in order to determine the resulting target labels once a metric has been passed through both relabeling configurations.    The way I understand it, the following steps could be taken:

1. Execute a request against  `select/{tenant}/prometheus/target-relabel-debug` to test the `relabel_configs` section of a job
2. Take the resulting labels of the previous step and use them as input labels against `select/{tenant}/prometheus/metric-relabel-debug` to test the `metric_relabel_configs` section?
3. The resulting labels would be exactly those that would be scraped by the job on vmagent.

Is this logic valid?

**Comment by zekker6** [comment]
@hartfordfive Yes, looks good to me. Just a quick note, don't forget to handle cases when target will be completely dropped by relabeling rules at stage 1 :)

**Comment by hartfordfive** [comment]
On a related note, would it be possible to have the web UI updated to include a input box for both `relabel_configs` and `metric_relabel_configs` so that's it's available in the same interface?   That would make the testing a bit easier rather than have two separate URLs.   Considering prometheus scrape job configurations often have both `relabel_configs` and `metric_relabel_configs` rules, this might be a bit more straight forward for testing via the web interface.

**Comment by zekker6** [comment]
@hagen1778 @dmitryk-dk What do you think about this?

**Comment by hagen1778** [comment]
Hey @zekker6 @hartfordfive 
From the code I see that `/target-relabel-debug` and `/metric-relabel-debug` are very alike. The difference is the following:
1. `/target-relabel-debug` will try to extract `__address__` into `instance` label before dropping all `__`-meta labels
2. `/metric-relabel-debug` will drop all `__`-meta labels except `__name__`
If this difference is insignificant for your tests, then you can alway use `/target-relabel-debug`, for example.

**Comment by hagen1778** [comment]
Closing this ticket as answered.
